### PR TITLE
New .pdfpc format

### PR DESCRIPTION
**Pull request by fnevgeny**
Although this started with intention to replace the current format of pdfpc files (as discussed in #536), the bulk of the changes is actually about the metadata structure. The idea is to put everything page/slide-oriented (notes, user-defined overlays; in the future there could be hidden slides (#157)) into the same list.

The issues that are dealt by this PR are: #239, #355, #367,  #552, #554 

Questions:
* Are we set about JSON? The other option is XML; other formats will require extra libraries. JSON is often criticized for not having comments and issues with multiline content (e.g., https://www.lucidchart.com/techblog/2018/07/16/why-json-isnt-a-good-configuration-language/), but since in our case editing is not expected to be done by a human, I don't think it is a showstopper.

* Do we keep the same (.pdfpc) extension? Currently, I assume yes, and backup old files before overwriting with the new format.

### CODE REVIEW

**Comment by fnevgeny** [comment]
I forgot to mention: support for the "included" note files is dropped. There is no way to do it in JSON natively, and I didn't want to make a crude hack. Also, I guess the original request for a separate note file was due to [cebe/pdfpc-latex-notes](https://github.com/cebe/pdfpc-latex-notes), but now that we have pdfpc.sty that supports full metadata functionality, it is probably no longer needed.

Also the docs are not updated yet.

**Review by AndreasBilke — COMMENTED** [review]
- I agree with your discussion about pro/cons of JSON. I also favor JSON (also for being better human readable)
- Backup before write JSON to the .pdfpc file is a fair compromise

In general I like the idea for having slide based config data (and not things like this weird list of slides numbers for the overlay specification)

**Comment by fnevgeny** [comment]
@AndreasBilke do you see some potential problems with the current format? E.g., something that might be predictably changed in the near future due to some feature addition etc? In principle, there is API versioning, but better to think about probable changes in advance. Also, aesthetics-wise, do you prefer key names in underscore_connected, lowerCamel, or UpperCamel style?

**Comment by AndreasBilke** [comment]
If I look through the new feature list I don't see any obvious problems.
For aesthetics: I would prefer lowerCamel case since it's a common style in
todays API (e.g.
https://google.github.io/styleguide/jsoncstyleguide.xml?showone=Property_Name_Format#Property_Name_Format)

**Comment by fnevgeny** [comment]
Well, the logic is: load .pdfpc (if exists), parse PDF, unconditionally overwriting everything present, then apply command line adjustments, and then anything via GUI during the run time. Save the resulting config (except for "native" notes) to .pdfpc on exit.

If a property is defined in PDF itself and later modified via command line or GUI, it will be saved to .pdfpc *but* effectively ignored next time. It is a minor inconvenience; I assume if one defined something via pdfpc.sty, he/she could easily change it if needed.

For reference, here is a summary of presentation options & metadata, and ways to define them:

|                    |         |         |         |              |          |        | 
|--------------------|---------|---------|---------|--------------|----------|--------| 
| Feature            | Default | pdfpcrc | PDF/XMP | Command line | Run-time | .pdfpc | 
| duration           |         |         | yes     | yes          |          | yes    | 
| start_time         |         |         | yes     | yes          |          | yes    | 
| end_time           |         |         | yes     | yes          |          | yes    | 
| last_minutes       | 5       |         | yes     | yes          |          | yes    | 
| notes_position     |         |         | yes     | yes          |          | yes    | 
| default_transition |         |         | yes     | yes          |          | yes    | 
| disable_markdown   | FALSE   |         | yes     | yes          |          | yes    | 
| end_slide          |         |         | yes     |              | yes      | yes    | 
| last_saved_slide   |         |         |         |              | yes      | yes    | 
| font_size          |         |         |         |              | yes      | yes    | 
| notes              |         |         | yes     |              | yes      | yes    | 
| forced_overlays    |         |         |         |              | yes      | yes    | 
| time_of_day        | FALSE   | yes     |         | yes          |          |        | 
| windowed_mode      | None    | yes     |         | yes          |          |        |

File: man/pdfpc.in [hunk 1]
```diff
@@ -337,20 +340,21 @@ many minutes you are overtime.
 .SS Notes
 
 .PP
-Textual notes can be displayed for each slide.  A few types of PDF annotations
-are understood by pdfpc and will be automatically imported and displayed (only
-their textual content, no formatting attributes are preserved).  The PDF
-annotations can be made using many PDF editors and even viewers.  These "native"
-PDF notes cannot be edited in pdfpc.
+Textual notes can be displayed for each slide.  pdfpc understands the Markdown
```
Comment (AndreasBilke) [inline man/pdfpc.in:343]: Maybe I missed something but this PR is independant from the previous Markdown feature? Do you just want to fix the man page in one go?

**Review by AndreasBilke — APPROVED** [review]

File: man/pdfpc.in (hunk 1 above)
Comment (fnevgeny) [reply man/pdfpc.in:343]: You are right. Strictly speaking, it should have been in the MD patchset. I noticed it when adding man page entry for `--note-format`.

**Pull request merged**